packages/flutter_tools/lib/src/version.dart

```diff
@@ -264,51 +264,17 @@ class FlutterVersion {
     ).run();
   }
 
-  /// The name of the temporary git remote used to check for the latest
-  /// available Flutter framework version.
-  ///
-  /// In the absence of bugs and crashes a Flutter developer should never see
-  /// this remote appear in their `git remote` list, but also if it happens to
-  /// persist we do the proper clean-up for extra robustness.
-  static const String _versionCheckRemote = '__flutter_version_check__';
-
   /// The date of the latest framework commit in the remote repository.
   ///
   /// Throws [VersionCheckError] if a git command fails, for example, when the
   /// remote git repository is not reachable due to a network issue.
   static Future<String> fetchRemoteFrameworkCommitDate(String branch) async {
-    await _removeVersionCheckRemoteIfExists();
     try {
-      await _run(<String>[
-        'git',
-        'remote',
-        'add',
-        _versionCheckRemote,
-        globals.flutterGit,
-      ]);
-      await _run(<String>['git', 'fetch', _versionCheckRemote, branch]);
-      return _latestGitCommitDate(
-        branch: '$_versionCheckRemote/$branch',
-      );
+      await _run(<String>['git', 'fetch', '--tags']);
+      return _gitCommitDate(gitRef: '@{u}');
     } on VersionCheckError catch (error) {
-      if (globals.platform.environment.containsKey('FLUTTER_GIT_URL')) {
-        globals.printWarning('Warning: the Flutter git upstream was overridden '
```

Comment:
Removed this message as the user will hit a similar error message when they try to upgrade, and in the case `FLUTTER_GIT_URL` is set incorrectly, the upstream of the current channel has to be exactly the same for the tool to arrive at this point.